lib/resque/worker.rb

```diff
@@ -244,6 +270,21 @@ def queues
       @queues.map {|queue| queue == "*" ? Resque.queues.sort : queue }.flatten.uniq
     end
 
+    # Returns a shuffled list of queues to use when searching for job.
+    # The order in which the queues appear is a combination of randomness and
+    # weigted factor assigned to each queue.
+    def shuffled_queues
+      return nil unless shuffling_enabled?
+
+      queues.map do |queue, i|
+        [rand() * @weights[queue], queue]
+      end.to_h.sort.reverse.to_h.values
+    end
+
+    def shuffling_enabled?
+      !(@weights.nil? || @weights.empty?)
```

Comment:
I do not do the former in order to do the latter.
Also, I was just following their *convention* in `validate_queues`.